mendeleev/models.py

```diff
@@ -657,29 +657,14 @@ def oxides(self) -> List[str]:
         return [f"{self.symbol}{cme}O{co}" for cme, co in normal_coeffs]
 
     def __hash__(self):
-        """Custom has function to allow comparisons
-
-        This drops allt he nested, related objects since SQLAlchemy use a custom
-        unhashable `InstrumentedList`.
-        """
-        to_drop = [
-            "_ionization_energies",
-            "_oxidation_states",
-            "_sa_instance_state",
-            "_series",
-            "_series_id",
-            "ec",
-            "group",
-            "ionic_radii",
-            "isotopes",
-            "screening_constants",
-        ]
-        hashable = [(k, v) for k, v in self.__dict__.items() if k not in to_drop]
-        return hash(tuple(sorted(hashable)))
+        # Allow Element objects as keys in dictionaries. Same atomic
+        # number means all other properties are equal too.
+        return self.atomic_number
```

Comment:
I'd personally be weary of hashing elements based on a single property, as it assumes there is no possible way for two elements to differ if their atomic numbers are the same. Because the `Element` class is mutable and can be used to generate custom, pesudo-/non-physical elements, it seems much more likely that there would be a collision based on this assumption than the birthday paradox on 32-bit architectures.

Reply:
I'm happy to remove the `__hash__` function until we make the object immutable. The hashable property wasn't part of https://github.com/lmmentel/mendeleev/issues/42 after all.

Reply:
I've made an alternative pull request for that, see https://github.com/lmmentel/mendeleev/pull/51